### ISSUE TITLE
Dynamically update translations using addResources at runtime?

### ISSUE THREAD

**Issue by BrianJVarley** [i18next/react-i18next#547]
*Overview*
Using react-i18next for existing translations needs. Came across a case to update the translation's values by key at runtime.

*Issue*
I've tried using https://www.i18next.com/overview/api#addresource addResource in this instance but it _adds_ a new key / value pair to the translations, rather than updates the existing value by key.

*Example:*

```
const runTimeTranslationUpdates = {"api.translations.engineTemp":  44.5}
i18next.addResources('en', 'translations', runTimeTranslationUpdates)

```
*Outcome*
In this case the translation expected to update by that given key, isn't changed in the render after addResources has been called.

*Expected Outcome*

addResources updates the existing translation's values by the given key `api.translations.engineTemp` in `runTimeTranslationUpdates` object.

*Question*

Is there an `i18next` api method to update translation on the fly by key?

I couldn't find an issue similar to this, hence why I've opened a new issue.
This issue is similar but is aimed at adding rather than updating the translations:
https://github.com/i18next/react-i18next/issues/356

**Comment by jamuhl** [comment]
https://www.i18next.com/overview/api#addresource -> API is like:

`i18next.addResources('en', 'translations', 'api.translations.engineTemp',  '44.5')`

**Comment by jamuhl** [comment]
ah you're using the addResources: https://github.com/i18next/i18next/blob/master/src/ResourceStore.js#L61 should work...but the value needs to be a string.

**Comment by BrianJVarley** [comment]
@jamuhl this is the API function I'm using addResources. But it isn't refreshing the translations. 

Should addResources do that or is there another method call needed to trigger the translates again?

I see this line emits the `added` event after addResources is called:

https://github.com/i18next/i18next/blob/0de7e5683565bcda8654aed5e9a59bf1ba916eaa/src/ResourceStore.js#L90

**Comment by BrianJVarley** [comment]
Turned out the i18nextConfig.ns value is being interpreted
as an array when passed to addResources.

So giving `i18next.addResources(lng, i18nextConfig.ns, runTimeData); doesn't work
as i18nextConfig.ns - `["translation"]` is an array.

The [transformOptions](https://github.com/i18next/i18next/blob/master/src/defaults.js#L66) function
does this transform.

Before:

```
export const i18nextConfig: any = {
  lng: "de",
  ns: "translation",
  resources: {
    de: {
      translation: {...{data: apiDataDeJson}}
    },
    en: {
      translation: {...{data: apiDataEnJson}}
    }
  }
};
export const runTimeTranslations = (runTimeData: any, lng: string) => {
  i18next.addResources(lng, i18nextConfig.ns, runTimeData);
};
export const i18n: i18next.i18n = i18next.init(i18nextConfig);

```

After: 
Fixed this by passing i18nextConfig.ns[0] as inputOption to `i18next.addResources`:
Maybe it should be called out in the docs, that `ns` option is mutated to an array type in init?

```

export const i18nextConfig: any = {
  lng: "de",
  ns: ["translation"],
  resources: {
    de: {
      translation: {...{data: apiDataDeJson}}
    },
    en: {
      translation: {...{data: apiDataEnJson}}
    }
  }
};
export const runTimeTranslations = (runTimeData: any, lng: string) => {
  i18next.addResources(lng, i18nextConfig.ns[0], runTimeData);
};
export const i18n: i18next.i18n = i18next.init(i18nextConfig);

```

**Comment by jamuhl** [comment]
<img width="689" alt="screen shot 2018-10-15 at 15 46 40" src="https://user-images.githubusercontent.com/977772/46954679-81201980-d091-11e8-8ce7-8d27c3f23a17.png">

**Comment by jamuhl** [comment]
could this be closed?

**Comment by ryapapap** [comment]
Wanted to leave a note for anyone in the future that might find this. I got stuck trying to figure out why this didn't work and had to dig pretty deep into the code to figure out that using ICU format (from i18next-icu) will 'cache' the old values and must also refresh on add. So, I needed to update my `init` config with:

```
i18nFormat: {
  bindI18nStore: "added",
},
react: {
  bindI18nStore: "added",
},
```

afterwards, calling `i18n.addResources` would work.
